bndtools.core.services/src/org/bndtools/core/sync/SynchronizeWorkspaceWithEclipse.java

```diff
@@ -114,32 +114,16 @@ private void sync(Collection<Project> doNotUse) {
 							if (!project.hasNature(Plugin.BNDTOOLS_NATURE))
 								continue;
 
-							System.out.println("remove " + project);
-							refresh = true;
-							after.accept("remove project", () -> WorkspaceSynchronizer.removeProject(project, monitor));
+							after.accept("remove project", () -> {
+								if (!monitor.isCanceled()) {
+									WorkspaceSynchronizer.removeProject(project, monitor);
+								}
+							});
 						}
 
 						if (monitor.isCanceled())
 							return null;
 
-						if (refresh) {
-
-							for (Project p : workspace.getAllProjects()) {
-
-								if (monitor.isCanceled())
-									return null;
-
-								p.clean();
-							}
-
-							after.accept("refresh", () -> {
-								root.refreshLocal(IResource.DEPTH_INFINITE, monitor);
```

Comment:
I think the disabling of autobuilding and the forcing of a clean and then full build are the big issues. Leave the refreshLocal call here.